Unikernel Deployment

giobart opened the pull request:
Added Unikernel deployment manual page. 
This covers:
- How to package your unikernel
- How to create an SLA for your unikernel
- How to enable Unikernel virtualization on your worker node

giobart committed:
unikernel operations

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
```diff
@@ -1,12 +1,134 @@
+
+## Packaging your first Unikraft image 
+
+Once you build your first Unikernel using [Unikraft](http://unikraft.org), you can create a tarball of the kernel image and the necessary files.
```
```suggestion
Once you build your first Unikernel using Unikraft, you can create a tarball of the kernel image and the necessary files.
```
Since it's linked a few lines above this is probably a bit superfluous

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
```diff
@@ -1,12 +1,134 @@
+Your unikernel tarball MUST contain a file named `kernel`, which is the unikernel image itself. Additionally, it CAN contain a `files1` folder, which is mounted by Oakestra as kernel filesystem at runtime. This folder contains all the additional files for the unikernel to run.
+
+{{< callout context="tip" title="Looking for an example?" icon="outline/rocket" >}}
+Check out the **Unikraft Nginx** example in the Application Catalogue section of the wiki.
```
```suggestion
Check out the **[Unikraft Nginx](../app-catalog/nginx-unikernel-deployment/)** example in the Application Catalogue section of the wiki.
```
This is definitely requires a link, but it won't work until this branch is rebased onto the app-catalog branch

giobart replied:
created an issue to fix this when that's merged. Broken links are harder to spot than missing ones :D

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
```diff
@@ -1,12 +1,134 @@
+You can then deploy the application as usual using the Oakestra Dashboard, the Oakestra CLI, or the Oakestra APIs.
+
+{{< callout context="caution" title="Caution" icon="outline/alert-triangle" >}}
+You need at least one worker node in one of your clusters with a unikernel runtime enabled to deploy your unikernels. 
```
```suggestion
You need at least one worker node in one of your clusters with a KVM runtime enabled to deploy your unikernels. 
```

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
```diff
@@ -1,12 +1,134 @@
+
+Oakestra supports the deployment of unikernels built using [Unikraft](http://unikraft.org). 
+
+#### What do you need?
```
I think this should be done in the style of the the other requirements callouts such as on the `oak-environment/high-level-setup-overview/` page

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
````diff
@@ -1,12 +1,134 @@
+ }
+ ```
+
+ #### What's important to note in this SLA?
````
```suggestion
 #### Note
```
I think this is a bit long

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
```diff
@@ -1,12 +1,134 @@
+The rest of the fields are similar to the ones used for container services.
+You can then deploy the application as usual using the Oakestra Dashboard, the Oakestra CLI, or the Oakestra APIs.
+
+{{< callout context="caution" title="Caution" icon="outline/alert-triangle" >}}
```
I think just `Caution` isn't that concise as a title

Mjaethers replied:
That being said, if the highlight this requirement at the beginning of the page, it might be a bit superfluous here

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
```diff
@@ -1,12 +1,134 @@
+If your node supports nested virtualization and you have KVM installed, you can enable the KVM runtime for your Oakestra worker node.
+
+{{< callout context="tip" title="How to install KVM?" icon="outline/rocket" >}}
+Please follow [this](https://phoenixnap.com/kb/ubuntu-install-kvm) guide to install KVM on your worker node.
```
I think we should mention that this is ubuntu specific

giobart replied:
```suggestion
E.g., If your worker node uses an Ubuntu distro, you can follow [this](https://phoenixnap.com/kb/ubuntu-install-kvm) guide to install KVM on your worker node.
```

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
```diff
@@ -1,12 +1,134 @@
+
+All you need to do is:
+
+1: Stop your worker node (if already running)
```
```suggestion
1. Stop your worker node (if already running)
```
In other places in the docs we use the simple `.` instead of a colon

Mjaethers replied:
```suggestion
1. Stop the node engine (if already running)
```

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
```diff
@@ -1,12 +1,134 @@
+Please follow [this](https://phoenixnap.com/kb/ubuntu-install-kvm) guide to install KVM on your worker node.
+{{< /callout >}}
+
+All you need to do is:
```
```suggestion
Follow these steps on the **worker node**:
```

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
````diff
@@ -1,12 +1,134 @@
+```bash
+sudo NodeEngine -a <Cluster Orchestrator IP> -d
+```
+{{< callout context="note" title="Note" icon="outline/info-circle" >}}
````
Maybe you could find a more concise title than `Note`?

Mjaethers reviewed (commented):
Some thoughts of mine. Keep in mind that I don't know the first thing on unikernels, so keep that in mind regarding any technical comments of mine

giobart committed:
Update content/docs/manuals/unikernel-operations.md

Co-authored-by: Matthew Humphreys <ge65yay@tum.de>

giobart committed:
Update content/docs/manuals/unikernel-operations.md

Co-authored-by: Matthew Humphreys <ge65yay@tum.de>

giobart committed:
Update content/docs/manuals/unikernel-operations.md

Co-authored-by: Matthew Humphreys <ge65yay@tum.de>

giobart committed:
Update content/docs/manuals/unikernel-operations.md

giobart committed:
Integrated Mjaethers' feedback

giobart committed:
Merge branch 'unikernel-manual' of https://github.com/nitindermohan/oak-new-docs into unikernel-manual

giobart committed:
KVM install callout fix

giobart committed:
typo

giobart committed:
typo

giobart commented:
@Mjaethers thanks a looot for the feedback. I think I've integrated almost everything you suggested.

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
````diff
@@ -1,12 +1,133 @@
+ `--...
+```
+
+Your unikernel tarball MUST contain a file named `kernel`, which is the unikernel image itself. Additionally, it CAN contain a `files1` folder, which is mounted by Oakestra as kernel filesystem at runtime. This folder contains all the additional files for the unikernel to run.
````
```suggestion
Your unikernel tarball MUST contain a file named `kernel`, which is the unikernel image itself. Additionally, it CAN contain a `files1` folder, which is mounted by Oakestra as a kernel filesystem at runtime. This folder contains all the additional files required by the unikernel to run.
```

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
```diff
@@ -1,12 +1,133 @@
+
+All you need to start deploying unikernels is:
+
+1. A unikernel .tar.gz hosted somewhere accessible to the Oakestra.
```
```suggestion
1. A unikernel `.tar.gz` hosted somewhere accessible to Oakestra
```

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
```diff
@@ -1,12 +1,133 @@
+
+1. A unikernel .tar.gz hosted somewhere accessible to the Oakestra.
+2. A service deployment descriptor 
+3. A Worker Node with a Unikernel Runtime enabled
```
```suggestion
3. A worker node with a unikernel runtime enabled
```

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
````diff
@@ -1,12 +1,133 @@
+```bash
+sudo NodeEngine stop
+```
+2: Enable unikernel runtime using 
````
```suggestion
2. Enable unikernel runtime using 
```

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
````diff
@@ -1,12 +1,133 @@
+```bash
+sudo NodeEngine config virtualization unikernel on
+```
+3: Re-start your worker node
````
```suggestion
3. Re-start your worker node
```

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
```diff
@@ -1,12 +1,133 @@
+
+## Packaging your first Unikraft image 
+
+Once you build your first Unikernel using Unikraft, you can create a tarball of the kernel image and the necessary files.
```
```suggestion
Once you build your first unikernel using Unikraft, you can create a tarball of the kernel image and the necessary files.
```

Mjaethers commented on content/docs/manuals/unikernel-operations.md:
````diff
@@ -1,12 +1,133 @@
+
+Once you build your first Unikernel using Unikraft, you can create a tarball of the kernel image and the necessary files.
+
+```bash
````
```suggestion
```yaml
```

giobart replied:
that’s actually the output of the `tree` bash command

Mjaethers reviewed (commented):
Looks nice! Some minor suggestions

giobart committed:
Update content/docs/manuals/unikernel-operations.md

Co-authored-by: Matthew Humphreys <ge65yay@tum.de>

giobart committed:
Apply suggestions from code review

Co-authored-by: Matthew Humphreys <ge65yay@tum.de>

nitindermohan committed:
unikernel updates

nitindermohan committed:
Merge remote-tracking branch 'origin/unikernel-manual' into unikernel-manual

Mjaethers commented:
It look like the last commit undid the changes I requested, could they be reapplied?

nitindermohan committed:
new text in multi-virtualizations

nitindermohan reviewed (changes requested):
Please see the updates and the new commit https://github.com/nitindermohan/oak-new-docs/commit/f5710be897ed32e207fd14460592c113c3bea025

nitindermohan commented:
@Mjaethers Sorry I overrode some of the requested changes as I updated the text. Please see if those changes are still required in the new updates

nitindermohan commented:
@giobart I added new text and changed the structure of the documentation. Please see if any changes are necessary and then we can merge this PR

giobart committed:
removed lxc and swapped docker with containerd

giobart commented:
> @giobart I added new text and changed the structure of the documentation. Please see if any changes are necessary and then we can merge this PR

Very cool the landing page. As of now, I have removed LXC containers from the list and added `containerd` instead for the container runtime. I guess we can add what we test over time.

nitindermohan reviewed (approved):
This looks good now. Ready to merge
